src/index.js

```diff
@@ -4,556 +4,582 @@ const codec = require('./codec')
+      return false
+    }
+    return true
   }
-  if (protos[1].code !== 6 && protos[1].code !== 273) {
-    return false
+
+  /**
+   * Creates a Multiaddr from a node-friendly address object
+   *
+   * @param {{family: string, address: string, port: string}} addr
+   * @param {string} transport
+   * @returns {Multiaddr} multiaddr
+   * @throws {Error} Throws error if addr or transport are not truthy
+   * @example
+   *
+   * ```js
+   * Multiaddr.fromNodeAddress({address: '127.0.0.1', port: '4001'}, 'tcp')
+   * // <Multiaddr 047f000001060fa1 - /ip4/127.0.0.1/tcp/4001>
+   * ```
+   */
+  static fromNodeAddress (addr, transport) {
+    if (!addr) { throw new Error('requires node address object') }
+    if (!transport) { throw new Error('requires transport protocol') }
+    let ip
+    switch (addr.family) {
+      case 'IPv4':
+        ip = 'ip4'
+        break
+      case 'IPv6':
+        ip = 'ip6'
+        break
+      default:
+        throw Error(`Invalid addr family. Got '${addr.family}' instead of 'IPv4' or 'IPv6'`)
+    }
+    return new Multiaddr('/' + [ip, addr.address, transport, addr.port].join('/'))
   }
-  return true
-}
 
-/**
- * Object containing table, names and codes of all supported protocols.
- * To get the protocol values from a Multiaddr, you can use
- * [`.protos()`](#multiaddrprotos),
- * [`.protoCodes()`](#multiaddrprotocodes) or
- * [`.protoNames()`](#multiaddrprotonames)
- *
- * @instance
- * @returns {{table: Array, names: Object, codes: Object}}
- *
- */
-Multiaddr.protocols = protocols
+  /**
+   * Returns if something is a Multiaddr that is a name
+   *
+   * @param {Multiaddr} addr
+   * @returns {boolean} isName
+   */
+  static isName (addr) {
+    if (!Multiaddr.isMultiaddr(addr)) {
+      return false
+    }
 
-/**
- * Returns if something is a Multiaddr that is a name
- *
- * @param {Multiaddr} addr
- * @returns {Bool} isName
- */
-Multiaddr.isName = function isName (addr) {
-  if (!Multiaddr.isMultiaddr(addr)) {
-    return false
+    // if a part of the multiaddr is resolvable, then return true
+    return addr.protos().some((proto) => proto.resolvable)
   }
 
-  // if a part of the multiaddr is resolvable, then return true
-  return addr.protos().some((proto) => proto.resolvable)
-}
-
-/**
- * Returns an array of multiaddrs, by resolving the multiaddr that is a name
- *
- * @async
- * @param {Multiaddr} addr
- * @returns {Multiaddr[]}
- */
-Multiaddr.resolve = function resolve (addr) {
-  if (!Multiaddr.isMultiaddr(addr) || !Multiaddr.isName(addr)) {
-    return Promise.reject(Error('not a valid name'))
+  /**
+   * Check if object is a CID instance
+   *
+   * @param {any} value
+   * @returns {value is Multiaddr}
+   */
+  static isMultiaddr (value) {
+    return value instanceof Multiaddr || Boolean(value && value[symbol])
   }
 
-  /*
-   * Needs more consideration from spec design:
-   *   - what to return
-   *   - how to achieve it in the browser?
+  /**
+   * Static factory method
+   *
```

Comment:
I think it would make sense to deprecate it to encourage transition.

```suggestion
   *
   * @deprecated
```